tests/test_bookmarks.py

```diff
@@ -0,0 +1,131 @@
+"""
+Test that the tap can replicate multiple pages of data
+"""
+import backoff
+from datetime import timedelta
+from singer.utils import strftime, strptime_to_utc
+from tap_tester import runner, menagerie
+from base import MambuBaseTest
+
+@backoff.on_predicate(backoff.expo, lambda x: x <= 0, max_tries=10)
+def poll_state_version(conn_id):
```

Comment:
Why are we polling? Is there something wrong with the state methods in tap-tester?

Reply:
Circle was hitting an error and it seems to be caused by the test trying to post a state before the sync does. So we make the test wait to post until something else has (which bumps the version up past 0)